omero_figure/views.py

```diff
@@ -369,12 +365,9 @@ def make_web_figure(request, conn=None, **kwargs):
     # If the figure has been saved, construct URL to it.
     figure_dict = json.loads(figure_json)
     if 'fileId' in figure_dict:
-        try:
-            figure_url = reverse('load_figure', args=[figure_dict['fileId']])
-            figure_url = request.build_absolute_uri(figure_url)
-            input_map['Figure_URI'] = wrap(figure_url)
-        except:
-            pass
+        figure_url = reverse('load_figure', args=[figure_dict['fileId']])
```

Comment:
No exception thrown anymore?

Reply:
We'd only get an exception here if a non-numeric value was set for 'fileId' in the JSON, in which case we probably want to know about it, rather than fail silently. I guess we could leave the try/except in place "just in case" since it's not critical and we wouldn't want figure export to fail just because of this? I don't have a strong view either way.

Reply:
If the id cannot be read then you cannot export surely so it should fail

Reply:
You can export to PDF/TIFF without needing to save (just tried it).

Reply:
maybe return an error indicating that the fileID was not correctly set so something meaningful is displayed in the activities dialog

Reply:
I think I'll just revert this change - we can't return an error AND start the export script running (which may complete OK). Existing behaviour is OK.